apollo-router-core/src/services/router_service.rs

```diff
@@ -308,14 +309,18 @@ impl PluggableRouterServiceBuilder {
         let (router_service, router_worker) = Buffer::pair(
             ServiceBuilder::new().service(
                 self.plugins.iter_mut().fold(
-                    RouterService::builder()
-                        .query_planner_service(query_planner_service)
-                        .query_execution_service(execution_service)
-                        .schema(self.schema)
-                        .query_cache(query_cache)
-                        .naive_introspection(naive_introspection)
-                        .build()
-                        .boxed(),
+                    // TODO: maybe make it optional although enabled by default?
+                    APQService::new(
+                        RouterService::builder()
+                            .query_planner_service(query_planner_service)
+                            .query_execution_service(execution_service)
+                            .schema(self.schema)
+                            .query_cache(query_cache)
+                            .naive_introspection(naive_introspection)
+                            .build(),
+                        512, // TODO: this is totally arbitrary, what does the team think?
```

Comment:
If we want to allow config of this then a plugin would be the way to go. I'm still not sure what we want to do with default plugins. It could be worth introducing a concept of a default plugin. APQ and Otel probably fall in to this category.
Let's create a separate ticket to follow up though.

Reply:
Worth noting that the gateway can build Apq layers using various cache implementations, which can then be set (for example a Memcached or anything).

However the default InMemoryLRUCache constructor [defaults to an infinite size]5ef7949d/packages/apollo-server-caching/src/InMemoryLRUCache.ts](https://github.com/apollographql/apollo-server/blob/52744b616847353d508b787e549dd4cc5ef7949d/packages/apollo-server-caching/src/InMemoryLRUCache.ts#L17) which is probably not a default we want in the router

Reply:
opened #440 440